### PR TITLE
Fix `bandit` high-severity warnings:

### PR DESCRIPTION
**Pull request by mbyrnepr2**
Remove `shell=True` argument of `subprocess.call`.

Refactor: Use `subprocess.run` instead of `subprocess.call`.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
|    | :bug: Bug fix          |
|    | :sparkles: New feature |
| ✓   | :hammer: Refactoring   |
|    | :scroll: Docs          |

## Description
Running `bandit` against the `pylint` repository yields 3 high-severity warnings.
This merge-request attempts to fix those.
In addition `subprocess.run` is used in favour of `subprocess.call`.

[Documentation](https://docs.python.org/3.11/library/subprocess.html#module-subprocess) regarding the `shell=True` argument for Windows:
_On Windows with shell=True, the COMSPEC environment variable specifies the default shell. The only time you need to specify shell=True on Windows is when the command you wish to execute is built into the shell (e.g. dir or copy). You do not need shell=True to run a batch file or console-based executable._

[Documentation](https://docs.python.org/3/library/subprocess.html#using-the-subprocess-module) regarding using `subprocess.run`:
_The recommended approach to invoking subprocesses is to use the [run()](https://docs.python.org/3/library/subprocess.html#subprocess.run) function for all use cases it can handle. For more advanced use cases, the underlying [Popen](https://docs.python.org/3/library/subprocess.html#subprocess.Popen) interface can be used directly._

[subprocess.call is considered part of the older API.](https://docs.python.org/3/library/subprocess.html#older-high-level-api)

<details>
<summary>Output of the bandit command:</summary>
(venv310) Marks-MacBook-Air-2:programming markbyrne$ bandit -r -lll pylint/pylint

[main]	INFO	profile include tests: None
[main]	INFO	profile exclude tests: None
[main]	INFO	cli include tests: None
[main]	INFO	cli exclude tests: None
[main]	INFO	running on Python 3.10.4
182 [0.. 50.. 100.. 150.. ]
Run started:2022-12-08 13:55:02.702590

Test results:
>> Issue: [B602:subprocess_popen_with_shell_equals_true] subprocess call with shell=True identified, security issue.
   Severity: High   Confidence: High
   CWE: CWE-78 (https://cwe.mitre.org/data/definitions/78.html)
   Location: pylint/pylint/graph.py:130:16
   More Info: https://bandit.readthedocs.io/en/1.7.4/plugins/b602_subprocess_popen_with_shell_equals_true.html
129	                    ],
130	                    shell=use_shell,
131	                )
132	            else:
133	                subprocess.call(
134	                    [self.renderer, "-T", target, dot_sourcepath, "-o", outputfile],
135	                    shell=use_shell,
136	                )
137	            os.unlink(dot_sourcepath)
138	        return outputfile
139	
140	    def emit(self, line: str) -> None:
141	        """Adds <line> to final output."""
142	        self.lines.append(line)
143	

--------------------------------------------------
>> Issue: [B602:subprocess_popen_with_shell_equals_true] subprocess call with shell=True identified, security issue.
   Severity: High   Confidence: High
   CWE: CWE-78 (https://cwe.mitre.org/data/definitions/78.html)
   Location: pylint/pylint/graph.py:135:16
   More Info: https://bandit.readthedocs.io/en/1.7.4/plugins/b602_subprocess_popen_with_shell_equals_true.html
134	                    [self.renderer, "-T", target, dot_sourcepath, "-o", outputfile],
135	                    shell=use_shell,
136	                )
137	            os.unlink(dot_sourcepath)
138	        return outputfile

--------------------------------------------------
>> Issue: [B602:subprocess_popen_with_shell_equals_true] subprocess call with shell=True identified, security issue.
   Severity: High   Confidence: High
   CWE: CWE-78 (https://cwe.mitre.org/data/definitions/78.html)
   Location: pylint/pylint/pyreverse/dot_printer.py:170:12
   More Info: https://bandit.readthedocs.io/en/1.7.4/plugins/b602_subprocess_popen_with_shell_equals_true.html
169	                ["dot", "-T", target, dot_sourcepath, "-o", outputfile],
170	                shell=use_shell,
171	            )
172	            os.unlink(dot_sourcepath)
173	

--------------------------------------------------

Code scanned:
	Total lines of code: 37969
	Total lines skipped (#nosec): 0

Run metrics:
	Total issues (by severity):
		Undefined: 0
		Low: 94
		Medium: 5
		High: 3
	Total issues (by confidence):
		Undefined: 0
		Low: 0
		Medium: 3
		High: 99
Files skipped (0):
</details>

### CODE REVIEW

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]
Great 👍 I think we could add bandit to the pre-commit configuration too (for high severity warnings only for now). https://github.com/PyCQA/bandit/blob/main/.pre-commit-hooks.yaml

**Review by DanielNoord — DISMISSED** [review]

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -132,6 +132,11 @@ repos:
         exclude: *fixtures
         args: ["--max-summary-lines=2", "--linewrap-full-docstring"]
         files: "pylint"
+  - repo: https://github.com/PyCQA/bandit
+    rev: 1.7.4
+    hooks:
+      - id: bandit
+        args: ["-r", "-lll"]
```
Comment (mbyrnepr2) [inline .pre-commit-config.yaml:139]:
```suggestion
```

File: tests/functional/a/access/access_attr_before_def_false_positive.py [hunk 2]
```diff
@@ -6,35 +6,25 @@
 self.cookedq."""
 
 
-import telnetlib
+class Tomato:
+    def __init__(self, juicy_index=42, deliciousness=100):
```
Comment (mbyrnepr2) [inline tests/functional/a/access/access_attr_before_def_false_positive.py:10]:
```suggestion
class Tomato:
    """The base Tomato class"""
    def __init__(self, juicy_index=42, deliciousness=100):
```

File: tests/functional/a/access/access_attr_before_def_false_positive.py [hunk 3]
```diff
@@ -6,35 +6,26 @@
...
+        self.juicy_index = juicy_index
+        self.deliciousness = deliciousness
 
-class SeeTelnet(telnetlib.Telnet):
-    """
-    Extension of telnetlib.
-    """
+    def print_description(self, label="generic"):
+        print(f"{label} tomato")
```
Comment (mbyrnepr2) [inline tests/functional/a/access/access_attr_before_def_false_positive.py:16]:
```suggestion
    def print_description(self, label="generic"):
        """Print important tomato-related information"""
        print(f"{label} tomato")
```

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -146,3 +146,8 @@ repos:
               pylint/__pkginfo__.py|
               setup.cfg
           )$
+  - repo: https://github.com/PyCQA/bandit
+    rev: 1.7.4
+    hooks:
+      - id: bandit
+        args: ["-r", "-lll"]
```
Comment (Pierre-Sassoulas) [inline .pre-commit-config.yaml:153]:
```suggestion
        args: ["-r", "-lll"]
        exclude: *fixtures
```

File: tests/functional/a/access/access_attr_before_def_false_positive.py [hunk 5]
```diff
@@ -6,35 +6,27 @@
 self.cookedq."""
 
 
-import telnetlib
```
Comment (Pierre-Sassoulas) [inline tests/functional/a/access/access_attr_before_def_false_positive.py:9]: I don't think we should modify the functional tests, could you revert this file please ?

File: tests/profile/test_profile_against_externals.py [hunk 6]
```diff
@@ -45,7 +46,7 @@ def test_run(tmp_path: Path, name: str, git_repo: str) -> None:
     """Runs pylint against external sources."""
     checkoutdir = tmp_path / name
     checkoutdir.mkdir()
-    os.system(f"git clone --depth=1 {git_repo} {checkoutdir}")
+    subprocess.run(["git", "clone", "--depth=1", git_repo, checkoutdir], check=True)
```
Comment (Pierre-Sassoulas) [inline tests/profile/test_profile_against_externals.py:49]: 👌 (We even have a  dependency to gitpython, we could even use it there.)

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]

**Review by Pierre-Sassoulas — APPROVED** [review]
👌

File: tests/functional/a/access/access_attr_before_def_false_positive.py (hunk 5 above)
Comment (mbyrnepr2) [reply tests/functional/a/access/access_attr_before_def_false_positive.py]: I expected this revert to fail pre-commit; your suggestion to exclude fixtures must have done the trick.
